internal/strace/strace.go

```diff
@@ -166,7 +165,9 @@ func (r *Result) recordFileWrite(file string, writeBuffer []byte, bytesWritten i
 	r.files[file].WriteInfo = append(r.files[file].WriteInfo, writeContentsAndBytes)
 	if _, exists := r.allWriteBufferId[writeID]; !exists {
 		if err := utils.CreateAndWriteTempFile(writeID, writeBuffer); err != nil {
-			log.Fatal("Could not create and write temp file", "error", err)
+			// TODO: push this error up to be handled by the caller.
+			logger.Error("Could not create and write temp file", "error", err)
```

Comment:
I think this should be logged to the main log (with context?), not the strace log

Reply:
Alright, then we may as well make the change to return it now

Reply:
Done